src/tools/fuse/README.md

```diff
@@ -0,0 +1,189 @@
+# README.md
+
+Mount [elektra](https://www.libelektra.org)'s KDB as a [fuse(py)](https://github.com/fusepy/fusepy) filesystem.
+
+This tool enables the inspection and modification of the KDB, in the form of a classical filesystem, as it is observed by a given running process.
+In the simplest case, an Elektra key appears as a file with the key name as file path and with the key value as file content.
+
+This enables users to interact with Elektra
+- using standard unix tooling, making Elektra more accessible
+- with standard system/library procedures
+- without the need to learn other tooling of libelektra
+- with additional features: e.g. the ability to effortlessly write binary keys
+
+## Installation and Quickstart
+
+### Docker
+
+The package comes with a preconfigured docker environment for expirimentation/debugging/developement.
+
+Ensure docker is installed and run
+```sh
+cd src/tools/fuse/docker
+./run_new_container.sh
+```
+This (re)creates the container if necessary and starts an interactive session with keys for testing already created (see `docker/create_keys.sh`).
+The filesystem is mounted below `~/mount`.
+For debugging purposes, a logfile is written to `~/nohup.out` inside the container. 
+
+### Native
+
+System requirements:
+- Elektra of version at least 0.9.5 with the python binding installed (for example, `libelektra5-all` satisfies this)
+- `fuse` (for example via the debian package `fuse`) 
+- `python3 >= 3.8`, `pip3`
+
+Required python3 packages (will be installed by `pip3` automatically if not present):
+- `fusepy>=3.0.1`
+- `psutil>=5.8.0`
+
+Build the `elektra_fuse` python3-wheel and install it using `pip3` with:
+```
+./build.sh && sudo ./install.sh
+```
+This, if needed, will also install the python-requirements `fusepy` and `psutil`.
+`install.sh` is run as root in order to make the new shell command `elektra_fuse` globally available (if run as a normal user, `$PATH` needs to be adapted as specified in the command's output).
```

Comment:
Better to install elektra_fuse to Elektra's tools (like the other tools), so that `kdb mount-fuse` (or similar) can be written.

Reply:
Would the following pose a proper way to integrate it?
- Hook into Elektras installation procedure and install the elektra_fuse python package to the global scope via pip3
- Expose it in kdb by having a new subcommand "mount-fuse" that essentially performs a subprocess call along the lines of "python3 -m elektra_fuse"

Reply:
>  Hook into Elektras installation procedure and install the elektra_fuse python package to the global scope via pip3

It is not simply anywhere globally but the CMake installation paths should also be respected. In particular, the installation should be below ${CMAKE_INSTALL_PREFIX}. As said, src/tools/pythongen/CMakeLists.txt already does something like that.

> Expose it in kdb by having a new subcommand "mount-fuse" that essentially performs a subprocess call along the lines of "python3 -m elektra_fuse"

Exactly, it is basically simply installing the script to ${CMAKE_INSTALL_PREFIX}/${TARGET_TOOL_EXEC_FOLDER}. Please also add metadata so that scripts/kdb/find-tools will work (see also doc/help/kdb-find-tools.md).

I would very much appreciate if you can document these steps in src/tools/README.md (as all tools should respect kdb-find-tools and install in CMake paths). And don't forget to also add the FUSE tool there.

Reply:
a32e9e9b554c43cf6f420280fcbebf6ef02603ac adds CMake integration satisfying these requirements, src/tools/README.md I will update next.

Reply:
I updated src/tools/README.md with information on the fuse tool and the required steps for new tools in 0df6a502ef3de4bad250ef74873060f522a297ff.

Reply:
What is the current state? Did you try to package the fuse binding?